src/IO/Base.agda

```diff
@@ -128,3 +128,8 @@ untilJust : IO (Maybe A) → IO A
 untilJust m = bind (♯ m) λ where
   nothing  → ♯ untilJust m
   (just a) → ♯ return a
+
+untilRight : ∀ {A B : Set} → (A → IO (A ⊎ B)) → A → IO B
```

Comment:
```suggestion
untilRight : (A → IO (A ⊎ B)) → A → IO B
```

This will:
* rely on the `variable` block instead of quantifying locally
* make the definition level polymorphic